### ISSUE TITLE
Pylint 2.3 causes AttributeError: 'MessagesStore' object has no attribute 'get_message_definition'

### ISSUE THREAD

**Issue by craiga** [PyCQA/pylint-plugin-utils#12]
It appears that the recently released Pylint 2.3 may have broken compatibility with this package.

```
$ pylint 
Traceback (most recent call last):
  File "/Users/craiga/.ve/env_mgmt_dashboard/bin/pylint", line 10, in <module>
    sys.exit(run_pylint())
  File "/Users/craiga/.ve/env_mgmt_dashboard/lib/python3.7/site-packages/pylint/__init__.py", line 20, in run_pylint
    Run(sys.argv[1:])
  File "/Users/craiga/.ve/env_mgmt_dashboard/lib/python3.7/site-packages/pylint/lint.py", line 1583, in __init__
    linter.load_plugin_modules(plugins)
  File "/Users/craiga/.ve/env_mgmt_dashboard/lib/python3.7/site-packages/pylint/lint.py", line 637, in load_plugin_modules
    module.register(self)
  File "/Users/craiga/.ve/env_mgmt_dashboard/lib/python3.7/site-packages/pylint_django/plugin.py", line 30, in register
    apply_augmentations(linter)
  File "/Users/craiga/.ve/env_mgmt_dashboard/lib/python3.7/site-packages/pylint_django/augmentations/__init__.py", line 763, in apply_augmentations
    suppress_message(linter, TypeChecker.visit_attribute, 'no-member', is_model_field_display_method)
  File "/Users/craiga/.ve/env_mgmt_dashboard/lib/python3.7/site-packages/pylint_plugin_utils/__init__.py", line 125, in suppress_message
    get_message_definition = msgs_store.get_message_definition
AttributeError: 'MessagesStore' object has no attribute 'get_message_definition'
```

**Comment by sphuber** [comment]
I can confirm this behavior. The combination `pylint-2.3.0 pylint-django-2.0.5 pylint-plugin-utils-0.4` is now broken whereas `pylint-2.2.2 pylint-django-2.0.5 pylint-plugin-utils-0.4` is working

**Comment by sphuber** [comment]
I am not too familiar with the internals, but it seems the version of `astroid` is also important. With `astroid==2.2.0` the linter still fails, even after downgrading to `pylint==2.2.2`, but this time there is no exception stack trace printed, but I just get multiple instances of:
```
.
  Line: None
    pylint: failure / Tool pylint failed to run (exception was raised)
```
Downgrading `astroid` to `2.1.0` fixes this problem as well and pylint runs again.
TLDR: `astrooid==2.2.0` with `pylint==2.3.0` seems to break

**Comment by craiga** [comment]
@sphuber That might be caused by https://github.com/PyCQA/astroid/issues/649, which looks like a separate issue to me.

**Comment by sphuber** [comment]
@craiga seems you are right, thanks for the reference

**Comment by shepilov-vladislav** [comment]
I did pull request for this https://github.com/PyCQA/pylint-plugin-utils/pull/13

**Comment by carlio** [comment]
@sphuber are you using prospector? If you are and if you have time, could you reproduce this error:

`Line: None
    pylint: failure / Tool pylint failed to run (exception was raised)`

but then run prospector again with the `--die-on-tool-error` to get a full stack trace?

**Comment by carlio** [comment]
Fixed with pylint-plugin-utils 0.5 release (and pylint-django 2.0.6 release)

**Comment by sphuber** [comment]
@carlio yeah it was being run with `prospector==1.1.5` indeed, although unfortunately I do not have the time to test it now. Thanks for the tip and the fix though!

**Comment by sphuber** [comment]
Ok it seems that `prospector` suffers from the same problem, because after running it with the flag you suggested I get the stack trace:
```
  File "/home/sphuber/.virtualenvs/aiida_py3/lib/python3.5/site-packages/prospector/tools/pylint/collector.py", line 21, in handle_message
    self.add_message(msg.msg_id, location, msg.msg)
  File "/home/sphuber/.virtualenvs/aiida_py3/lib/python3.5/site-packages/prospector/tools/pylint/collector.py", line 32, in add_message
    msg_data = self._message_store.get_message_definition(msg_id)
AttributeError: 'MessagesStore' object has no attribute 'get_message_definition'
```

**Comment by carlio** [comment]
@sphuber yeah - prospector users pylint-django which uses pylint-plugin-utils but prospector is currently pinned to pylint-django==2.0.2 so I will need to fix that too. thanks for the extra debug help!

**Comment by sphuber** [comment]
I ran this locally with `pylint-django==2.0.6` though and all other packages are up to date. Is this not just a problem of `prospector==1.1.5` itself?

**Comment by abumalick** [comment]
I ran into the same error message with `prospector==1.1.6.4`, `pylint==2.3.1`, `pylint-django==2.0.9`, `pylint-plugin-utils==0.5`, installed with pyenv.

```
prospector -X
Traceback (most recent call last):
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/bin/prospector", line 10, in <module>
    sys.exit(main())
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/prospector/run.py", line 173, in main
    prospector.execute()
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/prospector/run.py", line 66, in execute
    messages += tool.run(found_files)
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/prospector/tools/pylint/__init__.py", line 254, in run
    self._linter.check(self._args)
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/pylint/lint.py", line 943, in check
    self._do_check(files_or_modules)
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/pylint/lint.py", line 1075, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/pylint/lint.py", line 1158, in check_astroid_module
    walker.walk(ast_node)
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/pylint/utils.py", line 1303, in walk
    self.walk(child)
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/pylint/utils.py", line 1305, in walk
    cb(astroid)
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/pylint_plugin_utils/__init__.py", line 57, in augment_func
    augmentation(chain, node)
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/pylint_plugin_utils/__init__.py", line 148, in do_suppress
    chain()
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/pylint_plugin_utils/__init__.py", line 93, in __exit__
    self._linter.add_message(*to_append_args, **to_append_kwargs)
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/pylint/utils.py", line 496, in add_message
    message_definition, line, node, args, confidence, col_offset
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/pylint/utils.py", line 571, in add_one_message
    confidence,
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/prospector/tools/pylint/collector.py", line 21, in handle_message
    self.add_message(msg.msg_id, location, msg.msg)
  File "/Users/abumalick/.local/share/virtualenvs/my-project-8uVUZegj/lib/python3.7/site-packages/prospector/tools/pylint/collector.py", line 32, in add_message
    msg_data = self._message_store.get_message_definition(msg_id)
AttributeError: 'MessagesStore' object has no attribute 'get_message_definition'
```

**Comment by abumalick** [comment]
pipenv graph:
```
prospector==1.1.6.4
  - dodgy [required: >=0.1.9, installed: 0.1.9]
  - mccabe [required: >=0.5.0, installed: 0.6.1]
  - pep8-naming [required: >=0.3.3,<=0.4.1, installed: 0.4.1]
  - pycodestyle [required: >=2.0.0,<=2.4.0, installed: 2.4.0]
  - pydocstyle [required: >=2.0.0, installed: 3.0.0]
    - six [required: Any, installed: 1.12.0]
    - snowballstemmer [required: Any, installed: 1.9.0]
  - pyflakes [required: >=0.8.1,<2.0.0, installed: 1.6.0]
  - pylint [required: ==2.3.1, installed: 2.3.1]
    - astroid [required: >=2.2.0,<3, installed: 2.2.5]
      - lazy-object-proxy [required: Any, installed: 1.4.1]
      - six [required: Any, installed: 1.12.0]
      - typed-ast [required: >=1.3.0, installed: 1.4.0]
      - wrapt [required: Any, installed: 1.11.2]
    - isort [required: >=4.2.5,<5, installed: 4.3.21]
    - mccabe [required: >=0.6,<0.7, installed: 0.6.1]
  - pylint-celery [required: ==0.3, installed: 0.3]
    - astroid [required: >=1.0, installed: 2.2.5]
      - lazy-object-proxy [required: Any, installed: 1.4.1]
      - six [required: Any, installed: 1.12.0]
      - typed-ast [required: >=1.3.0, installed: 1.4.0]
      - wrapt [required: Any, installed: 1.11.2]
    - pylint [required: >=1.0, installed: 2.3.1]
      - astroid [required: >=2.2.0,<3, installed: 2.2.5]
        - lazy-object-proxy [required: Any, installed: 1.4.1]
        - six [required: Any, installed: 1.12.0]
        - typed-ast [required: >=1.3.0, installed: 1.4.0]
        - wrapt [required: Any, installed: 1.11.2]
      - isort [required: >=4.2.5,<5, installed: 4.3.21]
      - mccabe [required: >=0.6,<0.7, installed: 0.6.1]
    - pylint-plugin-utils [required: >=0.2.1, installed: 0.5]
      - pylint [required: Any, installed: 2.3.1]
        - astroid [required: >=2.2.0,<3, installed: 2.2.5]
          - lazy-object-proxy [required: Any, installed: 1.4.1]
          - six [required: Any, installed: 1.12.0]
          - typed-ast [required: >=1.3.0, installed: 1.4.0]
          - wrapt [required: Any, installed: 1.11.2]
        - isort [required: >=4.2.5,<5, installed: 4.3.21]
        - mccabe [required: >=0.6,<0.7, installed: 0.6.1]
  - pylint-django [required: ==2.0.9, installed: 2.0.9]
    - pylint [required: >=2.0, installed: 2.3.1]
      - astroid [required: >=2.2.0,<3, installed: 2.2.5]
        - lazy-object-proxy [required: Any, installed: 1.4.1]
        - six [required: Any, installed: 1.12.0]
        - typed-ast [required: >=1.3.0, installed: 1.4.0]
        - wrapt [required: Any, installed: 1.11.2]
      - isort [required: >=4.2.5,<5, installed: 4.3.21]
      - mccabe [required: >=0.6,<0.7, installed: 0.6.1]
    - pylint-plugin-utils [required: >=0.5, installed: 0.5]
      - pylint [required: Any, installed: 2.3.1]
        - astroid [required: >=2.2.0,<3, installed: 2.2.5]
          - lazy-object-proxy [required: Any, installed: 1.4.1]
          - six [required: Any, installed: 1.12.0]
          - typed-ast [required: >=1.3.0, installed: 1.4.0]
          - wrapt [required: Any, installed: 1.11.2]
        - isort [required: >=4.2.5,<5, installed: 4.3.21]
        - mccabe [required: >=0.6,<0.7, installed: 0.6.1]
  - pylint-flask [required: ==0.6, installed: 0.6]
    - astroid [required: >=1.0, installed: 2.2.5]
      - lazy-object-proxy [required: Any, installed: 1.4.1]
      - six [required: Any, installed: 1.12.0]
      - typed-ast [required: >=1.3.0, installed: 1.4.0]
      - wrapt [required: Any, installed: 1.11.2]
    - pylint [required: >=1.0, installed: 2.3.1]
      - astroid [required: >=2.2.0,<3, installed: 2.2.5]
        - lazy-object-proxy [required: Any, installed: 1.4.1]
        - six [required: Any, installed: 1.12.0]
        - typed-ast [required: >=1.3.0, installed: 1.4.0]
        - wrapt [required: Any, installed: 1.11.2]
      - isort [required: >=4.2.5,<5, installed: 4.3.21]
      - mccabe [required: >=0.6,<0.7, installed: 0.6.1]
    - pylint-plugin-utils [required: >=0.2.1, installed: 0.5]
      - pylint [required: Any, installed: 2.3.1]
        - astroid [required: >=2.2.0,<3, installed: 2.2.5]
          - lazy-object-proxy [required: Any, installed: 1.4.1]
          - six [required: Any, installed: 1.12.0]
          - typed-ast [required: >=1.3.0, installed: 1.4.0]
          - wrapt [required: Any, installed: 1.11.2]
        - isort [required: >=4.2.5,<5, installed: 4.3.21]
        - mccabe [required: >=0.6,<0.7, installed: 0.6.1]
  - pylint-plugin-utils [required: >=0.2.6, installed: 0.5]
    - pylint [required: Any, installed: 2.3.1]
      - astroid [required: >=2.2.0,<3, installed: 2.2.5]
        - lazy-object-proxy [required: Any, installed: 1.4.1]
        - six [required: Any, installed: 1.12.0]
        - typed-ast [required: >=1.3.0, installed: 1.4.0]
        - wrapt [required: Any, installed: 1.11.2]
      - isort [required: >=4.2.5,<5, installed: 4.3.21]
      - mccabe [required: >=0.6,<0.7, installed: 0.6.1]
  - pyyaml [required: Any, installed: 5.1.1]
  - requirements-detector [required: >=0.6, installed: 0.6]
    - astroid [required: >=1.4, installed: 2.2.5]
      - lazy-object-proxy [required: Any, installed: 1.4.1]
      - six [required: Any, installed: 1.12.0]
      - typed-ast [required: >=1.3.0, installed: 1.4.0]
      - wrapt [required: Any, installed: 1.11.2]
  - setoptconf [required: >=0.2.0, installed: 0.2.0]
```

**Comment by abumalick** [comment]
It seems not to be related to pylint-django and pylint-celery:

```
prospector --no-autodetect -X
Traceback (most recent call last):
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/bin/prospector", line 10, in <module>
    sys.exit(main())
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/prospector/run.py", line 173, in main
    prospector.execute()
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/prospector/run.py", line 66, in execute
    messages += tool.run(found_files)
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/prospector/tools/pylint/__init__.py", line 254, in run
    self._linter.check(self._args)
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/pylint/lint.py", line 943, in check
    self._do_check(files_or_modules)
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/pylint/lint.py", line 1075, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/pylint/lint.py", line 1158, in check_astroid_module
    walker.walk(ast_node)
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/pylint/utils.py", line 1303, in walk
    self.walk(child)
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/pylint/utils.py", line 1305, in walk
    cb(astroid)
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/pylint/checkers/variables.py", line 1790, in leave_functiondef
    super(VariablesChecker3k, self).leave_functiondef(node)
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/pylint/checkers/variables.py", line 1030, in leave_functiondef
    self._check_is_unused(name, node, stmts[0], global_names, nonlocal_names)
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/pylint/checkers/variables.py", line 964, in _check_is_unused
    "unused-argument", args=name, node=stmt, confidence=confidence
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/pylint/checkers/__init__.py", line 110, in add_message
    self.linter.add_message(msg_id, line, node, args, confidence, col_offset)
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/pylint/utils.py", line 496, in add_message
    message_definition, line, node, args, confidence, col_offset
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/pylint/utils.py", line 571, in add_one_message
    confidence,
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/prospector/tools/pylint/collector.py", line 21, in handle_message
    self.add_message(msg.msg_id, location, msg.msg)
  File "/Users/abumalick/.local/share/virtualenvs/fugoki-monolith-8uVUZegj/lib/python3.7/site-packages/prospector/tools/pylint/collector.py", line 32, in add_message
    msg_data = self._message_store.get_message_definition(msg_id)
AttributeError: 'MessagesStore' object has no attribute 'get_message_definition'
```

**Comment by sphuber** [comment]
No its just that `pylint` has changed the API of the `MessageStore` and `prospector` has not adapted to it yet

**Comment by abumalick** [comment]
Thank you very much @sphuber 
pinning prospector to `==1.1.6.2` as described in your liked issues fixed the problem

it is a bit strange that `prospector==1.1.6.4` require `pylint==2.3.1` while they are not compatible.
https://github.com/PyCQA/prospector/blob/master/setup.py#L38
